src/Modal/index.jsx

```diff
@@ -34,11 +33,12 @@ class Modal extends React.Component {
     if (this.firstFocusableElement) {
       this.firstFocusableElement.focus();
     }
-    this.parentElement = document.querySelector(this.props.parentSelector);
-    if (this.parentElement === null) {
-      throw new Error(`Modal recieved invalid parentSelector: ${this.props.parentSelector}, no matching element found`);
+    if (this.props.parentSelector) {
+      this.parentElement = document.querySelector(this.props.parentSelector);
+      if (this.parentElement === null) {
+        throw new Error(`Modal recieved invalid parentSelector: ${this.props.parentSelector}, no matching element found`);
+      }
     }
-    this.parentElement.appendChild(this.el);
```

Comment:
Another somewhat major change: the modal will render directly into `this.parentElement` instead of creating `this.el`, rendering the Modal into `this.el`, and then appending it to the `this.parentElement`. I didn't see any reason to have the extra `this.el` div so I removed it. Was there any reason for it that I am not aware of?